### ISSUE TITLE
fix(model): rewind live target on partial MTP transaction commit

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#12342]
Parent: #12235
Follow-on to closed #9988

## Classification
- **Centrality:** Core
- **Worker tier:** Astra contract/review; Sol implementation
- **Write set:** `internal/model/mtp_transaction.go`, `internal/model/mtp_transaction_test.go`

## For / Problem / Today / Better because / Witness
- **For:** callers binding `MTPTransaction` to a live native target session.
- **Problem:** after evaluating K proposals and accepting `0 < k < K`, the target must equal checkpoint plus exactly the first k proposals.
- **Today:** `Commit(k)` updates transaction bookkeeping and closes the snapshot without restoring the live session; rejected speculative tokens remain in decoder state.
- **Better because:** transaction accounting and live KV/recurrent state share the same committed boundary.
- **Witness:** `TestMTPTransactionPartialCommitRestoresLiveTargetPrefix`.

## Acceptance
- Prefill `[0,1]`, speculate `[2,3,4]`, execute the real target steps, then `Commit(1)`.
- Require target state and continuation logits to equal a fresh target consuming `[0,1,2]`.
- Require accepted/rejected accounting `1/2` and released checkpoint ownership.

## Witness command
`go test ./internal/model -run '^TestMTPTransactionPartialCommitRestoresLiveTargetPrefix$' -count=1`

## Non-goals
Verifier redesign, sampled acceptance, generation-budget truncation, or detached `MTPState` semantics.

**Comment by anthony-chaudhary** [comment]
Claimed: implementing the two-file live partial-commit leaf. Contract is checkpoint restoration followed by replay of only the accepted prefix, with exact native hybrid state/continuation and ownership/accounting witness. No verifier or generation-boundary changes.

**Comment by anthony-chaudhary** [comment]
Claiming the two-file transaction leaf. Contract: restore the pre-round live snapshot S and replay only D[:k], so Commit(k, StepK(S,D)) equals Stepk(S,D[:k]). Witness will execute a synthetic native hybrid target after prefill [0,1] and proposals [2,3,4], compare complete prefix state and continuation against a fresh target, and check checkpoint ownership plus accounting. No verifier or generation-boundary changes.

**Comment by anthony-chaudhary** [comment]
Host-native witness is green. Before the fix, the exact test failed with live positions [0 1 2 3 4] versus [0 1 2]. After checkpoint restoration plus accepted-prefix replay, the same test passes, including bit-exact KV/Kraw, recurrent, convolution, hidden, position/lineage state, one continuation step, 3/1/2 proposed/accepted/rejected accounting, and released checkpoint ownership. All TestMTPTransaction tests pass. Isolated fak validate passed gofmt, build, vet, and selected tests. Its automatic Strix phase was unreachable (SSH exit 255), adjudicated non-applicable to this host state-transition leaf, and tracked separately in #12382; no hardware evidence is claimed. The first guarded commit did not land: newer trunk has an unrelated containsStr redeclaration now tracked in #12383, and full-package validation timed out at four minutes. The two-file patch is preserved pending that repair, then guarded admission will be retried with a longer finite budget.

**Comment by anthony-chaudhary** [comment]
Claimed for the active DFlash/next-gen MTP goal under DOS lane \goal-dflash-mtp-12342\ (write set: \internal/model/mtp_transaction.go\, \internal/model/mtp_transaction_test.go\). Baseline audit confirms the named witness test is currently absent, so the first implementation step is a red reproduction; completion remains gated on \go test ./internal/model -run '^TestMTPTransactionPartialCommitRestoresLiveTargetPrefix$' -count=1\ plus path-scoped validation.

**Comment by anthony-chaudhary** [comment]
Shipped on trunk as \7a99b9dfb609bb604bac426f9385d4118c5a9476\ (\internal/model\): partial live-session MTP commits now restore the pre-round PrefixSnapshot and replay exactly the accepted draft prefix before cleanup. RED witness left cache/hidden history at 5 tokens instead of 3 after \Commit(1)\; GREEN: \go test ./internal/model -run '^TestMTPTransactionPartialCommitRestoresLiveTargetPrefix$' -count=1\. Path-scoped \ak validate --mine internal/model/mtp_transaction.go internal/model/mtp_transaction_test.go --smoke\ also passed, and \dos commit-audit 7a99b9dfb6\ returned witnessed.

**Comment by anthony-chaudhary** [comment]
Dependency unblocked: #12383 is closed and commit 726c52b64 is contained in origin/main. cmd/fak build, vet, and focused guard exposure tests passed.

**Comment by anthony-chaudhary** [comment]
Independently verified the shipped fix in [7a99b9dfb](https://github.com/anthony-chaudhary/fak/commit/7a99b9dfb609bb604bac426f9385d4118c5a9476), `internal/model@r710+g7a99b9dfb`.

- **Transition:** for an active live checkpoint and deterministic Step, restoring the checkpoint recovers S. After j accepted-token replays the state is Stepj(S,D[:j]); at j=k this proves Commit(k,StepK(S,D)) = Stepk(S,D[:k]) for 0<k<K.
- **Red → green:** the upstream exact test over parent `f5bb7469d` fails with cache/hidden lengths 5/3 and token history `[0 1 2 3 4]` versus `[0 1 2]`. On clean committed `a75b7802e` containing the fix, `go test ./internal/model -run '^TestMTPTransactionPartialCommitRestoresLiveTargetPrefix$' -count=1` passes (0.385s).
- **Coverage:** full hybrid cache and hidden-state equality, one continuation step/logits, accepted/rejected accounting 1/2, and released checkpoint ownership. All `TestMTPTransaction` tests and `go vet ./internal/model` pass.
- **Scope:** the additional automatic Strix validator phase was unreachable (SSH exit 255), adjudicated non-applicable to this host state-transition leaf; #12382 tracks relevance classification. No hardware evidence is claimed.

The upstream implementation already satisfies this issue. No redundant code commit was made; the two owned duplicate WIP paths were absorbed as `upstream_identical` through `fak wip park`, and the isolated checkout is clean. Closing #12342 on the witnessed upstream commit.

**Comment by anthony-chaudhary** [comment]
Unblocker #12383 landed on `origin/main` in 726c52b6491dd479ccdc3f123bedb3318234281b (`cmd/fak@r4066+g726c52b64`). The duplicate `containsStr` declaration is gone; focused cmd/fak vet, tests, and build are green. Please retry the guarded commit/validation for #12342.
